relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,78 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        // TODO: make keys lazy init
+        let mut keys = Vec::new();
```

Comment:
Vectors are lazy by design (see [docs](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new)):

> The vector will not allocate until elements are pushed onto it.

But I think it's good practice to only initialize the key on line 222. You could move the entire "response" block in there, because there is no need to run that when `keys` is empty.

Reply:
If they are lazy it is alright, otherwise `keys` isn't initiated yet in L222, if I declare it in L222, then it's not accessible in the other scope.
In Kotlin you can do `keys.isInitialized` before accessing to be sure, is that possible in Rust? Probably using `Option` instead.

Reply:
I used an `Option<BTreeSet<String>>` inited with `None`, let me know if there's a better way to avoid the `Option` or `BTreeSet` init overhead.